### PR TITLE
Master fix expense line ids sgv

### PR DESCRIPTION
**Pull request by sofiagvaladze**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_expense/models/hr_expense.py [hunk 1]
```diff
@@ -1003,6 +1004,25 @@ def _compute_from_employee_id(self):
...
+            # in draft employee can edit expense_line_ids
+            if report.employee_id.user_id == report.env.user and report.state == 'draft':
+                report.is_editable = True
+            elif report.state in ['draft', 'submit', 'approve']:
+                if is_manager:
+                    report.is_editable = True
+                else:
+                    # expense manager can edit, unless it's own expense
+                    current_managers = report.employee_id.expense_manager_id | report.employee_id.parent_id.user_id | report.employee_id.department_id.manager_id.user_id | report.user_id
+                    report.is_editable = (is_approver or report.env.user in current_managers) and report.employee_id.user_id != report.env.user
```
Comment (kba-odoo) [inline addons/hr_expense/models/hr_expense.py:1022]:
```suggestion
                    report.is_editable = (is_approver or self.env.user in current_managers) and report.employee_id.user_id != report.env.user
```

File: addons/hr_expense/security/ir_rule.xml [hunk 2]
```diff
@@ -37,18 +38,28 @@
...
                 ('employee_id.user_id', '=', user.id),
                 ('employee_id.department_id.manager_id.user_id', '=', user.id),
                 ('employee_id.parent_id.user_id', '=', user.id),
-                ('employee_id.expense_manager_id', '=', user.id)]</field>
+                ('employee_id.expense_manager_id', '=', user.id),
+                ('user_id', '=', user.id)]</field>
             <field name="groups" eval="[(4, ref('hr_expense.group_hr_expense_team_approver'))]"/>
         </record>
-        <record id="ir_rule_hr_expense_sheet_employee" model="ir.rule">
+        <record id="ir_rule_hr_expense_sheet_employee_draft" model="ir.rule">
```
Comment (kba-odoo) [inline addons/hr_expense/security/ir_rule.xml:49]: Keep the same ID, and it should be updated in a migration script:

```suggestion
        <record id="ir_rule_hr_expense_sheet_employee" model="ir.rule">
```

File: addons/hr_expense/security/ir_rule.xml [hunk 3]
```diff
@@ -37,18 +38,28 @@
...
             <field name="name">Employee Expense Sheet</field>
             <field name="model_id" ref="model_hr_expense_sheet"/>
-            <field name="domain_force">[('employee_id.user_id', '=', user.id)]</field>
+            <field name="domain_force">[('employee_id.user_id', '=', user.id), ('state', '=', 'draft')]</field>
+            <field name="groups" eval="[(4, ref('base.group_user'))]"/>
+        </record>
+        <record id="ir_rule_hr_expense_sheet_employee_rest" model="ir.rule">
```
Comment (kba-odoo) [inline addons/hr_expense/security/ir_rule.xml:55]:
```suggestion
        <record id="ir_rule_hr_expense_sheet_employee_not_draft" model="ir.rule">
```

**Review by kbapt — CHANGES_REQUESTED** [review]

**Comment by kba-odoo** [comment]
robodoo r+

**Comment by kba-odoo** [comment]
robodoo r+

**Pull request closed without merging**